File: Examples/VisualBasic/Working-with-Text-Boxes/SetShapeTextPositionAtLeft.vb
Imports Aspose.Diagram
Imports System
Public Class SetShapeTextPositionAtLeft
    Public Shared Sub Run()
        ' ExStart:SetShapeTextPositionAtLeft
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_ShapeTextBoxData()

        ' Load source Visio diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))
        ' Get shape
        Dim shapeid As Long = 1
        Dim shape As Shape = diagram.Pages.GetPage("Page-1").Shapes.GetShape(shapeid)

        ' Set text position at the left,
        ' TxtLocPinX = "TxtWidth*1" and TxtPinX = "Width*0"
        shape.TextXForm.TxtLocPinX.Value = shape.TextXForm.TxtWidth.Value
        shape.TextXForm.TxtPinX.Value = 0
        ' Set orientation angle
        Dim angleDeg As Double = 0
        Dim angleRad As Double = (Math.PI / 180) * angleDeg
        shape.TextXForm.TxtAngle.Value = angleRad

        ' Save Visio diagram in the local storage
        diagram.Save(dataDir & Convert.ToString("SetShapeTextPositionAtLeft_out.vsdx"), SaveFileFormat.VSDX)
        ' ExEnd:SetShapeTextPositionAtLeft
    End Sub
End Class

File: Examples/CSharp/Working-with-Text-Boxes/SetShapeTextPositionAtLeft.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.Working_with_Text_Boxes
{
    public class SetShapeTextPositionAtLeft
    {
        public static void Run() 
        {
            // ExStart:SetShapeTextPositionAtLeft
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ShapeTextBoxData();

            // Load source Visio diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");
            // Get shape
            long shapeid = 1;
            Shape shape = diagram.Pages.GetPage("Page-1").Shapes.GetShape(shapeid);

            // Set text position at the left,
            // TxtLocPinX = "TxtWidth*1" and TxtPinX = "Width*0"
            shape.TextXForm.TxtLocPinX.Value = shape.TextXForm.TxtWidth.Value;
            shape.TextXForm.TxtPinX.Value = 0;
            // Set orientation angle
            double angleDeg = 0;
            double angleRad = (Math.PI / 180) * angleDeg;
            shape.TextXForm.TxtAngle.Value = angleRad;

            // Save Visio diagram in the local storage
            diagram.Save(dataDir + "SetShapeTextPositionAtLeft_out.vsdx", SaveFileFormat.VSDX);
            // ExEnd:SetShapeTextPositionAtLeft
        }
    }
}
